static/js/actions/index.js

```diff
@@ -79,12 +82,14 @@ export const saveProfile = (username, profile) => {
       });
   };
 };
-export const updateProfileValidation = (username, errors) => ({
+
+export const UPDATE_PROFILE_VALIDATION = 'UPDATE_PROFILE_VALIDATION';
+export const updateProfileValidation = (username: string, errors: ValidationErrors): Action => ({
   type: UPDATE_PROFILE_VALIDATION,
   payload: { errors, username }
 });
 
-export function fetchUserProfile(username) {
+export function fetchUserProfile(username: string): Dispatcher {
   return dispatch => {
```

Comment:
does it make sense to add the same type checking here/`saveProfile`/etc as you did in `fetchDashboard` below? (referring to `return (dispatch: Dispatch) => {`)